File: src/BHS.Web/ClientApp/src/app/data/banners/models/alert-theme.ts
import { z } from 'zod';

export const alertThemeScheme = z.enum([
  'None',
  'Primary',
  'Secondary',
  'Success',
  'Danger',
  'Warning',
  'Info',
]);

export type AlertTheme = z.infer<typeof alertThemeScheme>;

File: src/BHS.Contracts/AlertTheme.cs
using System.Text.Json.Serialization;

namespace BHS.Contracts;

/// <summary>
/// The visual style of any notification, banner, or alert.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum AlertTheme : byte
{
    None = 0,
    Primary = 1,
    Secondary = 2,
    Success = 3,
    Danger = 4,
    Warning = 5,
    Info = 6,
}
